src/server/plugins/engine/configureEnginePlugin.ts

```diff
@@ -1,38 +1,45 @@
-import fs from 'node:fs'
-import { join } from 'node:path'
+import { join, parse } from 'node:path'
 
 import { type FormDefinition } from '@defra/forms-model'
 import { type ServerRegisterPluginObject } from '@hapi/hapi'
 
-import { idFromFilename } from '~/src/server/plugins/engine/helpers.js'
 import { FormModel } from '~/src/server/plugins/engine/models/FormModel.js'
 import {
   plugin,
   type PluginOptions
 } from '~/src/server/plugins/engine/plugin.js'
+import { type RouteConfig } from '~/src/server/types.js'
 
-export const configureEnginePlugin = (
-  formFileName?: string,
-  formFilePath?: string
-): ServerRegisterPluginObject<PluginOptions> => {
+export const configureEnginePlugin = async ({
+  formFileName,
+  formFilePath
+}: RouteConfig = {}): Promise<ServerRegisterPluginObject<PluginOptions>> => {
   let model: FormModel | undefined
 
   if (formFileName && formFilePath) {
-    const formConfigPath = join(formFilePath, formFileName)
+    const definition = await getForm(join(formFilePath, formFileName))
+    const { name } = parse(formFileName)
 
-    const definition = JSON.parse(
-      fs.readFileSync(formConfigPath, 'utf8')
-    ) as FormDefinition
-
-    const id = idFromFilename(formFileName)
-
-    model = new FormModel(definition, {
-      basePath: id
-    })
+    model = new FormModel(definition, { basePath: name })
   }
 
   return {
     plugin,
     options: { model }
   }
 }
+
+export async function getForm(importPath: string) {
+  const { ext } = parse(importPath)
+
+  const attributes: ImportAttributes = {
+    type: ext === '.json' ? 'json' : 'module'
+  }
+
+  const formImport = import(importPath, { with: attributes }) as Promise<{
```

Comment:
@colinrotherham do we need to gracefully handle any errors here if the import fails?

Reply:
Nah all good, we don't currently (see below) and we only load JSON files in tests

In production we load forms from **forms-manager** via API

https://github.com/DEFRA/forms-runner/blob/fadd272e674a2f369645dd01229b81d9ecee15c8/src/server/plugins/engine/configureEnginePlugin.ts#L23-L25

Reply:
ah okay, that makes sense.